src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -447,6 +447,143 @@ JSON example:
 }
 ```
 
+## Arterial Spin Labeling perfusion data
+
+Template:
+
+```Text
+sub-<label>/[ses-<label>/]
+    perf/
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_asl.nii[.gz]
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_asl.json
+       sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_aslcontext.tsv
+       [sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.nii[.gz]]
+       [sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_dir-<label>][_run-<index>]_m0scan.json]
+       [sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_labeling.jpg]
```

Comment:
I think the `deltam` and `cbf` suffixes should be represented in this template as well.

Reply:
We provided a table with volume types, which are often scanned sequentially in ASL. These can include any combination of raw volumes (control, label, and/or m0scan) and/or derivative volumes (deltam and/or cbf). To keep the original acquisition order intact (e.g. for subsequent motion correction and other temporally-specific artifacts) we opted to keep a single 4D NIfTI that contains any combination of volume types, which is then specified in the _asl_context.tsv sidecar. While deltam and cbf may indeed become separate derivative ASL NIfTIs, it may become confusing to also allow this as a suffix to the raw ASL NifTI. The reason to add the option to allow these derivatives deltam and cbf as raw volumes is that some vendor ASL sequences do not provide the original control-label raw data. In those cases, the derivatives deltam and cbf will (also) be seen as ‘raw’ data, as it is the only data available. Is this currently described with sufficient clarity?

Reply:
I think the logic for storing makes sense, but since they will still be valid files with their own rules regarding sets of entities and metadata, they should be reflected in a "Template". That template could possibly go below, where cbf and deltam are discussed in more detail, but I think it's just clearer to have them here with everything else.

Reply:
The files do not exist yet in this extension. Only in case there is no true raw data (control, label), the deltam or cbf can be used. However, even in this case, the file should always be called: *_asl.json. The distinction between control and label versus deltam or cbf in this case, is done in the aslcontext.tsv. The use of *_deltam or *_cbf should be supported in a (future) ASL-BIDS derivatives extension, so to avoid confusion, I think it is more logic not to mention it in the templates. Also, the validator will not yet agree with those files.

Reply:
Just to make sure I understand this, `cbf` and `deltam` would be volumes in the `_asl.nii.gz` file?

Reply:
Indeed. In most cases, the *_asl.nii will be one file with for example control-label-control-label-m0scan volumes, which you logically define in the aslcontext. Sometimes, additional files can be retrieved, such as a cbf image (calculated CBF so derivative), which should be stored in derivatives but is not yet included in this extension. However, some sequences (for example for GE) don't provide these raw data volumes, but already do the quantification online. So when getting your asl from the scanner, this will be for example one file with only a cbf volume. In this case (lack of true raw data), you can store your cbf image as *_asl.nii, with mentioning cbf as volume_type in the aslcontext. 

I hope this clarifies it?

Reply:
Yes, it does, thank you! I was misinterpreting your original explanation. In that case, you're totally right not to include nonexistent suffixes in the template here. I think the description of those volume types below could use some clarification, though, so others don't end up confused like me. I will bring that up at the relevant lines.